beacon-chain/blockchain/forkchoice/process_attestation.go

```diff
@@ -93,15 +96,31 @@ func (s *Store) OnAttestation(ctx context.Context, a *ethpb.Attestation) (uint64
 		return 0, err
 	}
 
-	// Use the target state to to validate attestation and calculate the committees.
-	indexedAtt, err := s.verifyAttestation(ctx, baseState, a)
-	if err != nil {
-		return 0, err
-	}
+	// Process aggregated attestation in the queue every `slot/2` duration,
+	// this allows the incoming attestation to aggregate and avoid
+	// excessively processing individual attestations.
+	fmt.Println(baseState.GenesisTime)
+	if halfSlot(baseState.GenesisTime) {
```

Comment:
Why wait until halfslot? what is this

Reply:
nvm i see the comment

Reply:
Instead of verifying attestation and update validator's vote one attestation after another. 
We aggregate attestations in a queue and process the queue after half slot. (Can modify this duration if we want)
With 1024 validators, before we were verifying 128 single attestation per slot, now we are verifying 1-2 attestations per slot. Huge saving